core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -1180,6 +1201,8 @@ sealed trait ZIO[-R, +E, +A] extends Serializable with ZIOPlatformSpecific[R, E,
    * Executes the effect on the specified `ExecutionContext` and then shifts back
    * to the default one.
    */
+  // TODO: Could we use the same `lock` name?
+  //  Instead of two different names for the same (somewhat advanced) method.
   final def on(ec: ExecutionContext): ZIO[R, E, A] =
```

Comment:
I think we have found overloading itself created issues, it is nice to say `for X to do A, for Y do B` rather than explain that there are actually two different variants of `A` that do different things.  We're moving away from that a little with `from` but I think that is quite explicitly saying "give me everything and I will turn it into a ZIO".